ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -284,7 +287,7 @@ def reset_env_if_ready(self, env: EnvManager, steps: int) -> None:
             and (self.resampling_interval)
             and (steps % self.resampling_interval == 0)
         )
-        if meta_curriculum_reset or generalization_reset:
+        if meta_curriculum_reset or generalization_reset or self.ghost_controller.reset:
```

Comment:
```suggestion
        ghost_controller_reset = self.ghost_controller.reset
        if meta_curriculum_reset or generalization_reset or ghost_controller_reset:
```
I think this is a little better if you were using both self-play and a curriculum or generalization. If both things wanted a reset, you'd reset two steps in a row  - the ghost controller wouldn't clear the flag because it would short-circuit the evaluation here. Unlikely to happen, but potentially annoying if it did.

Reply:
Done.